src/products/controller/products.controller.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  Put,
+} from '@nestjs/common';
+import { ProductsService } from '../service/products.service';
+import { ProductMapper } from '../mapper/product.mapper';
+import { CreateProductDTO } from '../dto/create-product.dto';
+import { Product } from '../domain/product.domain';
+import { ProductDTO } from '../dto/product.dto';
+import { ApiResponse, ApiTags } from '@nestjs/swagger';
+
+@ApiTags('products')
+@Controller('products')
+export class ProductsController {
+  constructor(
+    private readonly productService: ProductsService,
+    private productMapper: ProductMapper,
+  ) {}
+
+  @Post()
+  @ApiResponse({
+    status: 201,
+    description: 'The product has been successfully created.',
+  })
+  @ApiResponse({ status: 403, description: 'Forbidden.' })
+  async create(@Body() createProductDTO: CreateProductDTO): Promise<Product> {
+    return this.productService.createProduct(
+      this.productMapper.toDomain(createProductDTO),
+    );
+  }
+
+  @Get()
+  @ApiResponse({
+    status: 200,
+    description: 'All products retrieved successfully.',
+  })
+  @ApiResponse({ status: 403, description: 'Forbidden.' })
+  async findAll(): Promise<ProductDTO[]> {
+    const productCategories =
+      await this.productService.findAllProductCategories();
```

Comment:
Here some copy-paste issues. You are retrieving products, not product categories.